solidgm: build warnings on ppc64le

https://build.opensuse.org/build/home:wagi:branches:Base:System/openSUSE_Factory/ppc64le/nvme-cli/_log
```
abuild/rpmbuild/BUILD/nvme-cli-3.0_b.3-build/nvme-cli-3.0~b.3/ppc64le-suse-linux/nvme-config.h' -O2 -Wall -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=3 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -fstack-clash-protection -Werror=return-type -flto=auto -g -MD -MQ nvme.p/plugins_solidigm_solidigm-workload-tracker.c.o -MF nvme.p/plugins_solidigm_solidigm-workload-tracker.c.o.d -o nvme.p/plugins_solidigm_solidigm-workload-tracker.c.o -c ../plugins/solidigm/solidigm-workload-tracker.c
[   59s] ../plugins/solidigm/solidigm-workload-tracker.c: In function â€˜wltracker_show_newer_entriesâ€™:
[   59s] ../plugins/solidigm/solidigm-workload-tracker.c:381:31: warning: format â€˜%lluâ€™ expects argument of type â€˜long long unsigned intâ€™, but argument 2 has type â€˜__u64â€™ {aka â€˜long unsigned intâ€™} [-Wformat=]
[   59s]   381 |                 printf("%-16llu", timestamp);
[   59s]       |                         ~~~~~~^   ~~~~~~~~~
[   59s]       |                               |   |
[   59s]       |                               |   __u64 {aka long unsigned int}
[   59s]       |                               long long unsigned int
[   59s]       |                         %-16lu
[   59s] ../plugins/solidigm/solidigm-workload-tracker.c: In function â€˜wltracker_run_time_updateâ€™:
[   59s] ../plugins/solidigm/solidigm-workload-tracker.c:436:38: warning: format â€˜%lluâ€™ expects argument of type â€˜long long unsigned intâ€™, but argument 2 has type â€˜__u64â€™ {aka â€˜long unsigned intâ€™} [-Wformat=]
[   59s]   436 |                 printf("run_time: %lluus\n", wlt->run_time_us);
[   59s]       |                                   ~~~^       ~~~~~~~~~~~~~~~~
[   59s]       |                                      |          |
[   59s]       |                                      |          __u64 {aka long unsigned int}
[   59s]       |                                      long long unsigned int
[   59s]       |                                   %lu
[   59s] ../plugins/solidigm/solidigm-workload-tracker.c: In function â€˜sldgm_get_workload_trackerâ€™:
[   59s] ../plugins/solidigm/solidigm-workload-tracker.c:659:53: warning: format â€˜%lluâ€™ expects argument of type â€˜long long unsigned intâ€™, but argument 2 has type â€˜__u64â€™ {aka â€˜long unsigned intâ€™} [-Wformat=]
[   59s]   659 |                                 printf("Sleeping %lluus..\n", period_us);
[   59s]       |                                                  ~~~^         ~~~~~~~~~
[   59s]       |                                                     |         |
[   59s]       |                                                     |         __u64 {aka long unsigned int}
[   59s]       |                                                     long long unsigned int
[   59s]       |                                                  %lu
```

--- a/plugins/ocp/ocp-telemetry-decode.c
+++ b/plugins/ocp/ocp-telemetry-decode.c
@@ -780,7 +780,7 @@ int parse_time_stamp_event(
 			 SIZE_OF_VU_EVENT_ID));
 
 		ptime_stamp_event_vu_data =
-			(struct nvme_ocp_common_dbg_evt_class_vu_data *)((__u64)ptime_stamp_event +
+			(struct nvme_ocp_common_dbg_evt_class_vu_data *)((char *)ptime_stamp_event +
 			sizeof(struct nvme_ocp_time_stamp_dbg_evt_class_format));
 		vu_event_id = le16_to_cpu(ptime_stamp_event_vu_data->vu_event_identifier);
 		pdata = (__u8 *)&(ptime_stamp_event_vu_data->data);
@@ -850,7 +850,7 @@ int parse_pcie_event(
 			SIZE_OF_VU_EVENT_ID));
 
 		ppcie_event_vu_data =
-			(struct nvme_ocp_common_dbg_evt_class_vu_data *)((__u64)ppcie_event +
+			(struct nvme_ocp_common_dbg_evt_class_vu_data *)((char *)ppcie_event +
 			sizeof(struct nvme_ocp_pcie_dbg_evt_class_format));
 		vu_event_id = le16_to_cpu(ppcie_event_vu_data->vu_event_identifier);
 		pdata = (__u8 *)&(ppcie_event_vu_data->data);
@@ -919,7 +919,7 @@ int parse_nvme_event(
 			(sizeof(struct nvme_ocp_nvme_dbg_evt_class_format) +
 			SIZE_OF_VU_EVENT_ID));
 		pnvme_event_vu_data =
-			(struct nvme_ocp_common_dbg_evt_class_vu_data *)((__u64)pnvme_event +
+			(struct nvme_ocp_common_dbg_evt_class_vu_data *)((char *)pnvme_event +
 			sizeof(struct nvme_ocp_nvme_dbg_evt_class_format));
 
 		vu_event_id = le16_to_cpu(pnvme_event_vu_data->vu_event_identifier);
@@ -1032,7 +1032,7 @@ int parse_media_wear_event(
 			SIZE_OF_VU_EVENT_ID));
 
 		pmedia_wear_event_vu_data =
-			(struct nvme_ocp_common_dbg_evt_class_vu_data *)((__u64)pmedia_wear_event +
+			(struct nvme_ocp_common_dbg_evt_class_vu_data *)((char *)pmedia_wear_event +
 			sizeof(struct nvme_ocp_media_wear_dbg_evt_class_format));
 		vu_event_id = le16_to_cpu(pmedia_wear_event_vu_data->vu_event_identifier);
 		pdata = (__u8 *)&(pmedia_wear_event_vu_data->data);

--- a/plugins/solidigm/solidigm-workload-tracker.c
+++ b/plugins/solidigm/solidigm-workload-tracker.c
@@ -378,7 +378,7 @@ static int wltracker_show_newer_entries(struct wltracker *wlt)
 			timestamp += log->samplePeriodInMilliseconds * WLT2MS;
 			continue;
 		}
-		printf("%-16llu", timestamp);
+		printf("%-16" PRIu64, (uint64_t)timestamp);
 		for (int j = 0; j < MAX_FIELDS; j++) {
 			__u32 val = 0;
 			struct field f = group_fields[content_group][j];
@@ -433,7 +433,7 @@ void wltracker_run_time_update(struct wltracker *wlt)
 {
 	wlt->run_time_us = micros() - wlt->start_time_us;
 	if (nvme_args.verbose > 0)
-		printf("run_time: %lluus\n", wlt->run_time_us);
+		printf("run_time: %" PRIu64 "us\n", (uint64_t)wlt->run_time_us);
 }
 
 static int sldgm_stricmp(char const *a, char const *b)
@@ -656,7 +656,8 @@ int sldgm_get_workload_tracker(int argc, char **argv, struct command *acmd, stru
 			__u64 period_us = min(next_sample_us - wlt.run_time_us,
 					      stop_time_us - wlt.run_time_us);
 			if (nvme_args.verbose > 1)
-				printf("Sleeping %lluus..\n", period_us);
+				printf("Sleeping %" PRIu64 "us..\n",
+					(uint64_t)period_us);
 			usleep(period_us);
 			wltracker_run_time_update(&wlt);
 		}

